blog/2019-03-14-debezium-meets-quarkus.adoc

```diff
@@ -0,0 +1,193 @@
+
+[role="teaser"]
+--
+Last week's announcement of https://quarkus.io/[Quarkus] sparked a great amount of interest in the Java community:
+crafted from the best of breed Java libraries and standards, it allows to build Kubernetes-native applications based on GraalVM & OpenJDK HotSpot.
+In this blog post we are going to demonstrate how a Quarkus-based microservice can consume Debezium's data change events via Apache Kafka.
+For that purpose, we'll see what it takes to convert the shipment microservice from our recent post about the link:2019/02/19/reliable-microservices-data-exchange-with-the-outbox-pattern[outbox pattern] into Quarkus-based service.
+--
+
+Quarkus is a Java stack designed for the development of cloud-native applications based on the Java platform.
+It combines and tightly integrates mature libraries such Hibernate ORM, Vert.x, Netty, RESTEasy and Apache Camel as well as the APIs from the https://microprofile.io/[Eclipse MicroProfile] initiative,
+such as https://github.com/eclipse/microprofile-config[Config] or https://github.com/eclipse/microprofile-reactive-messaging[Reactive Messaging].
+Using Quarkus, you can develop applications using both imperative and reactive styles, also combining both approaches as needed.
+
+It is designed for significantly reduced memory consumption and improved startup time.
+Last but not least, Quarkus supports both OpenJDK HotSpot and GraalVM virtual machines.
+With GraalVM it is possible to compile the application into a native binary and thus reduce the resource consumption and startup time even more.
+
+To learn more about Quarkus itself, we recommend to take a look at its excellent https://quarkus.io/get-started/[Getting Started] guide.
+
+== Consuming Kafka Messages with Quarkus
+
+In the original https://github.com/debezium/debezium-examples/tree/master/outbox[example application] demonstrating the outbox pattern,
+there was a microservice ("shipment") based on Thorntail that consumed the events produced by the Debezium connector.
+We've extended the example with a new service named "shipment-service-quarkus".
+It provides the same functionality as the "shipment-service" but is implemented as a microservice based on Quarkus instead of Thorntail.
+
+This makes the overall architecture look like so:
+
+++++
+<div class="imageblock centered-image">
+    <img src="/images/outbox_pattern_quarkus.png" style="max-width:100%; margin-bottom:20px; margin-top:20px;" class="responsive-image" alt="Outbox Pattern Overview">
+</div>
+++++
+
+To retrofit the original service into a Quarkus-based application, only a few changes were needed:
+
+ * Quarkus right now supports only MariaDB but not MySQL; hence we have included an instance of MariaDB to which the service is writing
+ * The https://javaee.github.io/jsonp/[JSON-P API] used do deserialize incoming JSON messages is not supported yet; so the code has been modified to use the Jackson API which is already is supported by Quarkus
```

Comment:
So it's not exactly right JSON-P is not supported. It's supported in conjunction with RESTEasy but that's right we don't have a low-level library for it, and we should probably have one. I discussed this with Gunnar.